node/internal.ts

```diff
@@ -426,7 +427,7 @@ export function _which(tool: string, check?: boolean): string {
 function _tryGetExecutablePath(filePath: string, extensions: string[]): string {
     try {
         // test file exists
-        let stats: fs.Stats = fs.statSync(filePath);
+        let stats: fs.Stats = fs.statSync(_unTildify(filePath));
```

Comment:
also, within the past week 2 other people using the AZ CLI ran into this problem on macOS. It's fairly routine in `.bash_profile` to rely on `bash` support of `~` instead of something like `$HOME`